contrib/platform/src/com/sun/jna/platform/win32/User32.java

```diff
@@ -2484,5 +2487,329 @@ HANDLE SetWinEventHook(int eventMin, int eventMax, HMODULE hmodWinEventProc, Win
+	 *         even if several characters have been typed and are stored in the
+	 *         keyboard state. If possible, even with Unicode keyboard layouts, the
+	 *         function has written a spacing version of the dead-key character to
+	 *         the buffer specified by pwszBuff. For example, the function writes
+	 *         the character SPACING ACUTE (0x00B4), rather than the character
+	 *         NON_SPACING ACUTE (0x0301).</dd>
+	 *         <dt>0</dt>
+	 *         <dd>The specified virtual key has no translation for the current
+	 *         state of the keyboard. Nothing was written to the buffer specified by
+	 *         pwszBuff.</dd>
+	 *         <dt>1</dt>
+	 *         <dd>One character was written to the buffer specified by
+	 *         pwszBuff.</dd>
+	 *         <dt>2&le;value</dt>
+	 *         <dd>Two or more characters were written to the buffer specified by
+	 *         pwszBuff. The most common cause for this is that a dead-key character
+	 *         (accent or diacritic) stored in the keyboard layout could not be
+	 *         combined with the specified virtual key to form a single character.
+	 *         However, the buffer may contain more characters than the return value
+	 *         specifies. When this happens, any extra characters are invalid and
+	 *         should be ignored.</dd>
+	 *         </dl>
+	 * 
+	 */
+	int ToUnicodeEx(UINT wVirtKey, UINT wScanCode, byte[] lpKeyState, char[] pwszBuff, int cchBuff, UINT wFlags,
+			HKL dwhkl);
+
+	/**
+	 * Loads a string resource from the executable file associated with a specified
+	 * module, copies the string into a buffer, and appends a terminating null
+	 * character.
+	 * 
+	 * @param hInstance    A handle to an instance of the module whose executable
+	 *                     file contains the string resource. To get the handle to
+	 *                     the application itself, call the GetModuleHandle function
+	 *                     with NULL.
+	 * @param uID          The identifier of the string to be loaded.
+	 * @param lpBuffer     The buffer is to receive the string. Must be of
+	 *                     sufficient length to hold a pointer (8 bytes).
+	 * @param cchBufferMax The size of the buffer, in characters. The string is
+	 *                     truncated and null-terminated if it is longer than the
+	 *                     number of characters specified. If this parameter is 0,
+	 *                     then lpBuffer receives a read-only pointer to the
+	 *                     resource itself.
+	 * @return If the function succeeds, the return value is the number of
+	 *         characters copied into the buffer, not including the terminating null
+	 *         character, or zero if the string resource does not exist. To get
+	 *         extended error information, call GetLastError.
+	 * 
+	 */
+	int LoadString(HINSTANCE hInstance, UINT uID, LPSTR lpBuffer, int cchBufferMax);
```

Comment:
'lpBuffer` is `LPTSTR` and thus changes depending whether or not the unicode version is called. I would use a `Pointer` and add a convinience function to decode it. Have a look at this:
com.sun.jna.platform.win32.Kernel32Util.expandEnvironmentStrings(String)
there memory is allocated based on characters and read based on the unicode settings.

Reply:
Pointer - OK.
There is already a convenience function using the more common way to call this function.